test/e2e/README.md

```diff
@@ -0,0 +1,211 @@
+
+1. 
+    ```bash
+    make e2e-test ginkgo
+    ```
+2. 
+   ```bash
+   ./_bin/e2e-test run-e2e -f="al23-amd64 && simpleflow && ssm" --skip-cleanup=true --artifacts-dir=e2e-artifacts --logs-bucket=<s3 logs bucket> 
+   ```
+
+3. Passing --skip-cleanup allows for quickly reruning additional test cases without having to wait for new cluster creation/teardown.
+
+4. After running `run-e2e` to create your test infrastructure, you could use the following when rerunning:
+   ```bash
+    ./_bin/ginkgo  -v -tags=e2e --label-filter='al23-amd64 && simpleflow && ssm'  ./test/e2e/suite -- -filepath=e2e-artifacts/configs/e2e-param.yaml
+   ```
+
+### Key Options
+
+- `-f, --test-filter`: Filter tests using ginkgo label filters (e.g., "al23-amd64 && simpleflow && ssm")
+- `--skip-cleanup`: Skip cleanup after test completion
+- `--artifacts-dir`: Directory to store test artifacts (defaults to a temporary directory)
+- `--cni`: CNI plugin to use (default: "cilium")
+- `--logs-bucket`: S3 bucket for uploading test logs
+- `--no-color`: Disable colored output
+- `-n, --name`: Cluster name (default: "nodeadm-e2e-tests-1-31")
+- `-r, --region`: AWS region (default: "us-west-2")
+- `-k, --kubernetes-version`: Kubernetes version (default: "1.31")
+- `-p, --procs`: Number of processes to run (default: 1)
+- `--timeout`: Test timeout (default: "60m")
+- `--setup-config`: Path to YAML file containing cluster.TestResources configuration
+- `--test-config`: Path to YAML file containing suite.TestConfig configuration
+- `--ginkgo-binary`: Path to the ginkgo binary (defaults to binary in same folder as e2e-test or in PATH)
+
+
+You can run the tests manually using the CLI commands:
+
+1. **make**:
+    ```bash
+    make e2e-test ginkgo
+    ```
+
+2. **Setup the test infrastructure**:
+   ```bash
+   ./_bin/e2e-test setup -s path/to/e2e-config.yaml
+   ```
+
+3. **Run a single test**:
+   ```bash
+    ./_bin/ginkgo  -v -tags=e2e --label-filter='al23-amd64 && simpleflow && ssm'  ./test/e2e/suite -- -filepath=path/to/e2e-param.yaml
+   ```
+
+4. **Clean up the infrastructure**:
+   ```bash
+   # Using the resources file (for specific cleanup)
+   ./_bin/e2e-test cleanup -f path/to/resources.yaml
+    ```
+
+## Configuration Files
+
+### Cluster Resources Configuration (e2e-config.yaml)
+
+This file defines the infrastructure to be created for testing:
+
+```yaml
+clusterName: nodeadm-e2e-tests-1-31
+clusterRegion: us-west-2
+clusterNetwork:
+  vpcCidr: 10.0.0.0/16
+  publicSubnetCidr: 10.0.10.0/24
+  privateSubnetCidr: 10.0.20.0/24
+hybridNetwork:
+  vpcCidr: 10.1.0.0/16
+  publicSubnetCidr: 10.1.1.0/24
+  privateSubnetCidr: 10.1.2.0/24
+  podCidr: 10.2.0.0/16
+kubernetesVersion: "1.31"
+cni: cilium
+endpoint: ""
+```
+
+* cni - cilium or calico
+* endpoint - optional, intended to be used for teseting against beta or other environments
+
+
+### Test Parameters Configuration (e2e-param.yaml)
```

Comment:
Not sure how relevant it is but I just combine both the configs into a single yaml file so I only have to worry about the one file. We don't do strict un-marshal so whatever fields aren't needed are just ignored.